app/crud/labels.py

```diff
@@ -0,0 +1,271 @@
+import snowflake.snowpark.exceptions
+from snowflake.snowpark import Row
+from pydantic import (
+    validator,
+    root_validator,
+)
+from typing import Optional, ClassVar
+import datetime
+from .base import BaseOpsCenterModel
+from .session import session_ctx
+
+## TODO
+# migrate - maybe we don't move out of sql?
+# copy predefined labels to labels - maybe we dont move out of sql?
+# same again for probes
+# hook up to stored procs for (CRUD and validation)
+# more tests
+# ensure create, prepopulate, validate get called at the right times in bootstrap
+class Label(BaseOpsCenterModel):
+    table_name: ClassVar[str] = "LABELS"
+    on_success_proc: ClassVar[str] = "INTERNAL.UPDATE_LABEL_VIEW"
```

Comment:
ahh, it looks like you are already calling this in the subclass...why have both the on_success_proc and the lines (39, 43, 66)?

Reply:
> why have both the on_success_proc and the lines (39, 43, 66)

I meant to remove it from the Label class and forgot, heh. Intention was to have child classes declare what their "on success" procedure is, and have the parent class automatically call it for them.